### DISCUSSION TITLE
Save and Load template on Remote Storage

### DISCUSSION THREAD

**Discussion by manoxs** [GrapesJS/grapesjs#4567] [Q&A]
Hi there,

I'm in the process of updating an app to the latest version and there is a few breaking changes, with Storage Manager.
This is the update version of the config.

`storageManager: {
        type: 'remote',
        autoload: false,
        stepsBeforeSave: 3,
        options: {
            remote: {
                urlLoad: CompGrapesjsStore,
                urlStore: CompGrapesjsStore,
                contentTypeJson: true,
                storeComponents: true,
                storeStyles: true,
                storeHtml: true,
                onStore: (data, editor) => {
                    return { id: 1, data };
                },
                onLoad: result => result.data,
            }
        }
    },
`
The app has a custom command to save on the DB.
` commandsManager.add('saveToDatabase', {
        run: function (editor, sender) {
            editor.store();
            sender && sender.set('active');
            jQuery.notify('Template Saved!', 'success')
        },
    });`

Is there any more information for loading and storing template on remote storage?
Right now I'm now able to load the template once they are stored on the DB. 

I went thru the documentation and the the issue queue and this [4371](https://github.com/artf/grapesjs/issues/4371#issuecomment-1206388290) is the more helpful for loading remote data with backwards compatibility.

**Comment by artf** [comment]
You can follow the [docs here](https://grapesjs.com/docs/modules/Storage.html#setup-remote-storage) for setting up remote storage properly.

**Reply by manoxs** [reply]
Thanks for the reply! I'm wondering if there is any advice for resolving this same issue #[Import and export templates](https://github.com/artf/grapesjs/issues/2791#top) using remote storage and the latest API. 

I'm currently able to load and store in db, the issue is when I'm trying to load a predefined template.
The editor doesn't load until I reload the page.


```
document.addEventListener("DOMContentLoaded",function() {
    var done = null;

    if (typeof grapejs.editor !== 'undefined') {
        done = function () {
            grapejs.editor.load();
        };
    }
    var templateRevertLink = AjaxModal($('.navbar-app-menu a.revert_template'), {
        done: done,
        saved_message: "Template reset!"
    });

    //                          some templates don't have a RevertLink
    if (ShowTemplateSelector && templateRevertLink.length > 0) {
        templateRevertLink.magnificPopup('open');
    }

    AjaxModal($('.navbar-app-menu a.edit_settings'), {
        form_selector: "#edit-competition-form",
        saved_message: "Settings saved!",
        // modal: true
    });
});

```

**Reply by artf** [reply]
I guess you should rely on [loadProjectData](https://grapesjs.com/docs/modules/Storage.html#project-data) in that case

**Reply by manoxs** [reply]
Thanks! it works now
